robot_framework/get_organized/get_organized_api.py

```
@@ -0,0 +1,178 @@
+"""Functions for accessing the Miralix API."""
+
+import json
+
+from requests import Session
+from requests_ntlm import HttpNtlmAuth
+from robot_framework import config
+
+
+def create_session(apiurl: str, username: str, password: str) -> Session:
+    """Create a session for accessing GetOrganized API.
+
+    Args:
+        apiurl: URL for the API.
+        username: Username for login.
+        password: Password for login.
+
+    Returns:
+        Return the session object
+    """
+    session = Session()
+    session.headers.setdefault("Content-Type", "application/json")
+    session.auth = HttpNtlmAuth(username, password)
+    session.post(apiurl, timeout=config.GO_TIMEOUT)
+    return session
+
+
+def upload_document(*, apiurl: str, session: Session, file: bytearray, case: str, filename: str, agent_name: str | None = None, date_string: str | None = None) -> tuple[str, Session]:
+    """Upload a document to Get Organized.
+
+    Args:
+        apiurl: Base url for API.
+        session: Session token for request.
+        file: Bytearray of file to upload.
+        case: Case name already present in GO.
+        filename: Name of file when saved in GO.
+        agent_name: Agent name, used for creating a folder in GO. Defaults to None.
+        date_string: A date to add as metadata to GetOrganized. Defaults to None.
+
+    Returns:
+        Return response text and session token.
+    """
+    url = apiurl + "/_goapi/Documents/AddToCase"
+    payload = {
+        "Bytes": file,
+        "CaseId": case,
+        "SiteUrl": f"{apiurl}/cases/EMN/{case}",
+        "ListName": "Dokumenter",
+        "FolderPath": agent_name,
+        "FileName": filename,
+        "Metadata": f"<z:row xmlns:z='#RowsetSchema' ows_Dato='{date_string}'/>",
+        "Overwrite": True
+    }
+    response = session.post(url, data=json.dumps(payload), timeout=config.GO_TIMEOUT)
```

Should we raise an HTTP error here?

---

Same question for all requests